### PR TITLE
[Decoupling] Remove all references to coupled blobs

### LINKED ISSUE #6833: [DECOUPLING] Remove old containers/schemas
Remove `BlobsSidecar`, `SignedBlockAndBlobsSidecar` containers/schemas

### PR DESCRIPTION
**Pull request by StefanBratanov**
## PR Description
Since the decoupled block/blobs production is way different than the coupled version, It would be cleaner to remove current implementation, which in turn enables us to delete all coupled blob related containers.

Also modified BlockPublisherDeneb to have the `BlobSidecarPool` and the `BlobSidecarGossipChannel` which would be used to gossip/import blobs when implementing

## Fixed Issue(s)
fixes #6833 

## Documentation

- [X] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [X] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: beacon/validator/src/main/java/tech/pegasys/teku/validator/coordinator/publisher/MilestoneBasedBlockPublisher.java [hunk 1]
```diff
@@ -39,7 +40,8 @@ public MilestoneBasedBlockPublisher(
       final BlockFactory blockFactory,
       final BlockImportChannel blockImportChannel,
       final BlockGossipChannel blockGossipChannel,
-      final BlockAndBlobsSidecarGossipChannel blockAndBlobsSidecarGossipChannel,
+      final BlobSidecarPool blobSidecarPool,
+      final BlobSidecarGossipChannel blobSidecarGossipChannel,
```
Comment (zilm13) [inline beacon/validator/src/main/java/tech/pegasys/teku/validator/coordinator/publisher/MilestoneBasedBlockPublisher.java:44]: I think we should make it all the way `Optional` as there is no NOOP version

File: data/serializer/src/property-test/java/tech/pegasys/teku/provider/JsonProviderPropertyTest.java [hunk 2]
```diff
@@ -420,40 +417,4 @@ void roundTripKZGCommitment(@ForAll final int seed) throws JsonProcessingExcepti
...
-            .orElseThrow()
-            .getSignedBeaconBlockAndBlobsSidecarSchema();
-    assertThat(
-            deserialized.asInternalSignedBeaconBlockAndBlobsSidecar(
-                signedBeaconBlockAndBlobsSidecarSchema, spec))
-        .isEqualTo(
-            original.asInternalSignedBeaconBlockAndBlobsSidecar(
-                signedBeaconBlockAndBlobsSidecarSchema, spec));
-  }
 }
```
Comment (zilm13) [inline data/serializer/src/property-test/java/tech/pegasys/teku/provider/JsonProviderPropertyTest.java:420]: We could add in nice to have in decoupling issue to make it for BlobSidecar and other new structures probably

File: ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/execution/ExecutionPayloadResult.java [hunk 3]
```diff
@@ -51,7 +51,7 @@ public Optional<SafeFuture<HeaderWithFallbackData>> getExecutionPayloadHeaderFut
     return executionPayloadHeaderFuture;
   }
 
-  public Optional<SafeFuture<BlobsBundle>> getBlobsBundleFuture() {
+  public Optional<SafeFuture<BlobsBundle>> getBlobsBundle() {
```
Comment (zilm13) [inline ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/execution/ExecutionPayloadResult.java]: I think we need some consistency here, in this exact class. I'm ok with both versions, all future or all w/o future including vars

**Review by zilm13 — COMMENTED** [review]
Looks good for me, I would just add all removed tests like property tests in nice to have section of decoupled issue for new types and also rename following packages:
```
      tech.pegasys.teku.spec.logic.versions.deneb.blobs -> tech.pegasys.teku.spec.logic.versions.deneb.blob
      tech.pegasys.teku.beacon.sync.gossip.blobs -> tech.pegasys.teku.beacon.sync.gossip.blob
      tech.pegasys.teku.spec.datastructures.blobs -> tech.pegasys.teku.spec.datastructures.blob
      tech.pegasys.teku.spec.propertytest.suppliers.blobs -> tech.pegasys.teku.spec.propertytest.suppliers.blob
      tech.pegasys.teku.statetransition.blobs -> tech.pegasys.teku.statetransition.blob

      and for consistency:
      tech.pegasys.teku.beacon.sync.gossip.blocks -> tech.pegasys.teku.beacon.sync.gossip.block
      tech.pegasys.teku.spec.propertytest.suppliers.blocks -> tech.pegasys.teku.spec.propertytest.suppliers.block
```

File: ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/execution/ExecutionPayloadResult.java (hunk 3 above)
Comment (StefanBratanov) [reply ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/execution/ExecutionPayloadResult.java]: thanks, i changed them all to future suffix

File: data/serializer/src/property-test/java/tech/pegasys/teku/provider/JsonProviderPropertyTest.java (hunk 2 above)
Comment (StefanBratanov) [reply data/serializer/src/property-test/java/tech/pegasys/teku/provider/JsonProviderPropertyTest.java:420]: added one

File: beacon/validator/src/main/java/tech/pegasys/teku/validator/coordinator/publisher/MilestoneBasedBlockPublisher.java (hunk 1 above)
Comment (StefanBratanov) [reply beacon/validator/src/main/java/tech/pegasys/teku/validator/coordinator/publisher/MilestoneBasedBlockPublisher.java:44]: I created NOOP version, think it's cleaner than optional

**Comment by StefanBratanov** [comment]
> Looks good for me, I would just add all removed tests like property tests in nice to have section of decoupled issue for new types and also rename following packages:
> 
> ```
>       tech.pegasys.teku.spec.logic.versions.deneb.blobs -> tech.pegasys.teku.spec.logic.versions.deneb.blob
>       tech.pegasys.teku.beacon.sync.gossip.blobs -> tech.pegasys.teku.beacon.sync.gossip.blob
>       tech.pegasys.teku.spec.datastructures.blobs -> tech.pegasys.teku.spec.datastructures.blob
>       tech.pegasys.teku.spec.propertytest.suppliers.blobs -> tech.pegasys.teku.spec.propertytest.suppliers.blob
>       tech.pegasys.teku.statetransition.blobs -> tech.pegasys.teku.statetransition.blob
> 
>       and for consistency:
>       tech.pegasys.teku.beacon.sync.gossip.blocks -> tech.pegasys.teku.beacon.sync.gossip.block
>       tech.pegasys.teku.spec.propertytest.suppliers.blocks -> tech.pegasys.teku.spec.propertytest.suppliers.block
> ```

We have property tests for new blob structures. I added a task for package renaming but if it is part of this PR, like 300 files will be changed, so could do this as separate PR I think

**Review by zilm13 — APPROVED** [review]
LGTM

**Pull request merged**